### PR TITLE
feat(ext/http): implement serveHttp on serve [WIP]

### PR DESCRIPTION
**Pull request by mmastrac**
DO NOT MERGE

Re-implements `Deno.serveHttp` on top of `Deno.serve`.

Breaking changes:

 - `close` on `HttpConn` is now `async` requiring the caller to `await`. We need this because we cannot guarantee the underlying `Deno.serve` server has shut down fully. This was also implicitly async in the previous design.
 - `respondWith` no longer returns a promise that resolves when the connection has been closed or the connection returns an error. This functionality is missing in `Deno.serve` and may be replaced at a later time.
 - Removing the undocumented, unstable `http` namespace on `Deno`.

TODO:

 - [ ] Add test to Deno.serve for http+unix + host
 - [x] Add test to Deno.serve for AbortController::abort after shutdown

### CODE REVIEW

File: runtime/js/90_deno_ns.js [hunk 1]
```diff
@@ -133,11 +133,10 @@ const denoNs = {
   permissions: permissions.permissions,
   Permissions: permissions.Permissions,
   PermissionStatus: permissions.PermissionStatus,
-  // TODO(bartlomieju): why is this not in one of extensions?
```
Comment (mmastrac) [inline runtime/js/90_deno_ns.js:136]: Fixed this.

File: runtime/js/90_deno_ns.js [hunk 2]
```diff
@@ -199,9 +198,6 @@ denoNsUnstableById[unstableIds.fs] = {
 denoNsUnstableById[unstableIds.http] = {
   HttpClient: httpClient.HttpClient,
   createHttpClient: httpClient.createHttpClient,
-  // TODO(bartlomieju): why is it needed?
```
Comment (mmastrac) [inline runtime/js/90_deno_ns.js:202]: Removed this -- was not able to find a single reference to it.

File: cli/tests/unit/http_test.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ Deno.test(
 );
 
 Deno.test(
-  { permissions: { net: true } },
```
Comment (mmastrac) [inline cli/tests/unit/http_test.ts:114]: @bartlomieju Will need help going through these ignored tests to make sure we are not missing a major issue.

File: cli/tests/testdata/run/unstable_http.js [hunk 4]
```diff
@@ -2,12 +2,6 @@ const scope = import.meta.url.slice(-7) === "#worker" ? "worker" : "main";
 
 console.log(scope, Deno.HttpClient);
 console.log(scope, Deno.createHttpClient);
-console.log(scope, Deno.http?.HttpConn);
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply cli/tests/testdata/run/unstable_http.js:5]: That makes sense. I think this is a mistake in a test anyway.

**Comment by bartlomieju** [comment]
Let's close this for now. I think we're gonna be stuck with the current implementation.

**Pull request closed without merging**